### ISSUE TITLE
could it be pageable for searching

### ISSUE THREAD

**Issue by 10ngui** [github-tools/github#562]
#367
it is slow due to search for all items ,

**Comment by j-rewerts** [comment]
Yeah makes sense. It looks like we call the [Requestable._requestAllPages](https://github.com/github-tools/github/blob/master/lib/Search.js#L54) function. If we used [this](https://github.com/github-tools/github/blob/master/lib/Requestable.js#L156) function, you could pass through a page token as an option. 

I'm open to a PR!

**Comment by j-rewerts** [comment]
@WJXHenry, @kylejeske Let me know if you want to take this on.

**Comment by WJXHenry** [comment]
@j-rewerts Sure, I'll be happy to take this on!
